modules/claims_api/app/services/claims_api/dependent_claimant_poa_assignment_service.rb

```diff
@@ -51,19 +53,28 @@ def assign_poa_to_dependent_via_manage_ptcpnt_rlnshp?
         return true
       end
 
-      log(level: :warn,
-          detail: 'Something else went wrong with manage_ptcpnt_rlnshp. Falling back to update_benefit_claim.')
-
-      false
+      raise ::Common::Exceptions::ServiceError
     rescue ::Common::Exceptions::ServiceError => e
       if e.errors.first.detail == 'PtcpntIdA has open claims.'
         log(detail: 'Dependent has open claims, continuing.')
-      else
-        log(level: :warn,
-            detail: 'Something else went wrong with manage_ptcpnt_rlnshp. Falling back to update_benefit_claim.')
+
+        return false
       end
 
-      false
+      raise e
+    rescue => e
+      log(level: :error, detail: 'Something else went wrong with manage_ptcpnt_rlnshp.', error: error_details(e))
+
+      raise e
+    end
+
+    def error_details(e)
+      {
+        message: e.message,
+        detail: e.try(:detail),
+        details: e.try(:details),
+        errors: e.try(:errors)&.map(&:to_h)
+      }.compact
```

Comment:
@tycol7 How confident are we that e.message will not error. If this is a response from BGS, there is one thing I know about BGS, the responses are inconsistent. What about e&.message, e.try(:message)?

Reply:
[message](https://ruby-doc.org/core-3.0.0/Exception.html#method-i-message) is defined on the Exception object in ruby, so it should always be present here.